### PR TITLE
bin: Allow sideloading of lxd binaries

### PR DESCRIPTION
**Pull request by markylaing**
Adds a new environment variable `ENABLE_SIDELOADING` which stops the LXD daemon, then starts it again when the user presses enter. This allows time for loading a custom build.

I found this useful in my local testing.

### CODE REVIEW

**Comment by tomponline** [comment]
> Adds a new environment variable `ENABLE_SIDELOADING` which stops the LXD daemon, then starts it again when the user presses enter. This allows time for loading a custom build.
> 
> I found this useful in my local testing.

Good idea. But how about use LXD_SIDELOAD_PATH and update the scripts to copy that path binary into the correct location and reload LXD, avoiding the need for manual actions?

**Comment by markylaing** [comment]
> > Adds a new environment variable `ENABLE_SIDELOADING` which stops the LXD daemon, then starts it again when the user presses enter. This allows time for loading a custom build.
> > I found this useful in my local testing.
> 
> Good idea. But how about use LXD_SIDELOAD_PATH and update the scripts to copy that path binary into the correct location and reload LXD, avoiding the need for manual actions?

Yep sounds good. Will update.

**Comment by markylaing** [comment]
Have updated to use `LXD_SIDELOAD_PATH`.

File: bin/helpers [hunk 1]
```diff
@@ -128,6 +128,12 @@ install_lxd() (
     # on first invocation
     mkdir -p ~/snap/lxd/common/config/
     touch ~/snap/lxd/common/config/config.yml
+
+    if [ -n "${LXD_SIDELOAD_PATH:-}" ]; then
+        systemctl stop snap.lxd.daemon
+        cp "${LXD_SIDELOAD_PATH}" /var/snap/lxd/common/lxd.debug
+        systemctl start snap.lxd.daemon
+    fi
```
Comment (simondeziel) [inline bin/helpers:136]: I'd propose doing this but before the `waitready` earlier in that function. In the microcloud tests, we do it slightly differently so maybe you'd like to reuse it, https://github.com/canonical/microcloud/blob/main/test/includes/microcloud.sh#L136-L142

Comment (tomponline) [reply bin/helpers:136]: you can also just do cp, followed by reload rather than stop and start

**Pull request merged**